sys/net/crosslayer/ng_netreg/ng_netreg.c

```diff
@@ -101,4 +102,24 @@ ng_netreg_entry_t *ng_netreg_getnext(ng_netreg_entry_t *entry)
     return entry;
 }
 
+int ng_netreg_calc_csum(ng_pktsnip_t *hdr, ng_pktsnip_t *pseudo_hdr)
+{
+    switch (hdr->type) {
+#ifdef MODULE_NG_ICMPV6
+        case NG_NETTYPE_ICMPV6:
+            return (pseudo_hdr) ? ng_icmpv6_calc_csum(hdr, pseudo_hdr) : -EINVAL;
```

Comment:
keep it simple, just do the `if (pseudo_hdr == NULL) { return -EINVAL; }` at the beginning of the fuction. Or compile both solutions and show that this notation acutally saves memory (which I say it does the contrary) :-)